landlab/components/river_bed_dynamics/river_bed_dynamics.py

```diff
@@ -0,0 +1,3433 @@
+        ... 0, 1., 1., 1., 0,
+        ... 0, 1., 1., 1., 0,
+        ... 0, 1., 1., 1., 0,
+        ... 0, 1., 1., 1., 0,
+        ... 0, 1., 1., 1., 0,])
+
+        >>> gsd = np.array([[32, 100, 100], [16, 25, 50], [8, 0, 0]])
+
+        >>> timeStep = 1 # time step in seconds
+
+        >>> rbd = river_bed_dynamics(grid, gsd = gsd, dt = timeStep,
+        ... bedload_equation = 'Parker1990',
+        ... bed_surface__grain_size_distribution_location_node = gsd_loc)
+        >>> rbd.run_one_step()
+        >>> rbd.format_gsd(rbd._sediment_transport__bedload_grain_size_distribution_link)
+                   8         16     32
+        Link_0   0.0   0.000000    0.0
+        Link_1   0.0  51.520878  100.0
+        Link_2   0.0  51.520878  100.0
+        Link_3   0.0   0.000000    0.0
+        Link_4   0.0   0.000000    0.0
+        Link_5   0.0   0.000000    0.0
+        Link_6   0.0   0.000000    0.0
+        Link_7   0.0   0.000000    0.0
+        Link_8   0.0   0.000000    0.0
+        Link_9   0.0   0.000000    0.0
+        Link_10  0.0  52.498142  100.0
+        Link_11  0.0  52.498142  100.0
+        Link_12  0.0   0.000000    0.0
+        Link_13  0.0  45.944616  100.0
+        Link_14  0.0  71.774474  100.0
+        Link_15  0.0  52.498142  100.0
+        Link_16  0.0  71.774474  100.0
+        Link_17  0.0  45.944616  100.0
+        Link_18  0.0   0.000000    0.0
+        Link_19  0.0  71.774474  100.0
+        Link_20  0.0  71.774474  100.0
+        Link_21  0.0   0.000000    0.0
+        Link_22  0.0   0.000000    0.0
+        Link_23  0.0  71.774474  100.0
+        Link_24  0.0  71.774474  100.0
+        Link_25  0.0  71.774474  100.0
+        Link_26  0.0   0.000000    0.0
+        Link_27  0.0   0.000000    0.0
+        Link_28  0.0  71.774474  100.0
+        Link_29  0.0  71.774474  100.0
+        Link_30  0.0   0.000000    0.0
+        Link_31  0.0   0.000000    0.0
+        Link_32  0.0   0.000000    0.0
+        Link_33  0.0   0.000000    0.0
+        Link_34  0.0   0.000000    0.0
+        Link_35  0.0   0.000000    0.0
+        Link_36  0.0   0.000000    0.0
+        Link_37  0.0   0.000000    0.0
+        Link_38  0.0   0.000000    0.0
+        Link_39  0.0   0.000000    0.0
+
+        It also works for nodes.
+
+        """
+
+        if bedload_gsd.shape[0] == self._grid.number_of_links:
+            indexText = "Link_"
+        else:
+            indexText = "Node_"
+
+        bedload_gsd = np.hstack((bedload_gsd, np.zeros([bedload_gsd.shape[0], 1])))
+        bedload_gsd = np.cumsum(np.fliplr(bedload_gsd), axis=1) * 100
+
+        D_ascending = np.sort(self._grain_size_D_original)
+        columns = ["".join(item) for item in D_ascending.astype(str)]
+
+        df = pd.DataFrame(
+            bedload_gsd,
+            columns=columns,
+            index=[indexText + str(i) for i in range(bedload_gsd.shape[0])],
+        )
+        return df
+
+    @staticmethod
+    def display_available_fields():
```

Comment:
Instead of printing the available fields, how about just returning a list of the fields?

Reply:
Thanks, done! it is cleaner now.